system/timezone.go

```diff
@@ -3,31 +3,80 @@ package system
 import (
 	"log"
 	"os"
+	"path"
+	"regexp"
 )
 
-// GetTimezone returns the current system time zone
-func GetTimezone() (string, error) {
-	link, err := os.Readlink("/etc/localtime")
+// ReadTimezones returns a list of possible time zones
+// from the system
+// Possible arguments for zoneInfoDir:
+//	"" (root dir)
+//	"US"
+//	"posix/America"
+func ReadTimezones(zoneInfoDir string) (listZones []string, err error) {
+
+	file, err := os.Open(path.Join(zoneInfoPath, zoneInfoDir))
+	if err != nil {
+		log.Println("Error opening time zone file, ", err)
+		return nil, err
+	}
+
+	fileInfo, err := file.Readdir(-1)
+	if err != nil {
+		log.Println("Error reading time zones, ", err)
+		return nil, err
+	}
+
+	for _, fi := range fileInfo {
+		if !fi.IsDir() { // if file, not directory
+			listZones = append(listZones, fi.Name())
+		}
+	}
+
+	return listZones, nil
+}
+
+// GetTimezone returns the current system timezone and
+// its path after zoneInfoPath
+func GetTimezone() (zoneInfoDir, zone string, err error) {
+
+	link, err := os.Readlink(zoneLink)
 	if err != nil {
 		log.Println("Error finding time zone, ", err)
-		return "", err
+		return "", "", err
 	}
 
-	return link, nil
+	// extract the timezone and the zoneInfoDir from the full path
```

Comment:
@cbrake I used two regular expressions here in order to have the flexibility to extract either a file at the top level or a file with 1+ directories in its path.